packages/earljs/src/test-runners/uvu.ts

```diff
@@ -0,0 +1,65 @@
+import debug from 'debug'
+import { Callback, Test } from 'uvu'
+
+import { setTestRunnerIntegration } from '../testRunnerCtx'
+import { TestRunnerCtx } from './TestRunnerCtx'
+
+const d = debug('earljs:uvu')
+
+function main() {
+  if (typeof require !== 'function') {
+    return;
+  }
+
+  d('earljs/uvu integration is being registered...')
+
+  const uvu: typeof import('uvu') = require('uvu')
+  patchTest(uvu.test)
+
+  const suite = uvu.suite
+  uvu.suite = (...args) => {
+    const test = suite(...args)
+    patchTest(test)
+    return test
+  }
+}
+
+function patchTest(test: Test) {
+  const run = test.run.bind(test)
+  test.run = () => {
+    const testRunnerCtx: TestRunnerCtx = {
+      testInfo: {
+        suitName: [],
+        testName: '',
+        testFilePath: parseFilePathFromStackTrace(test.run),
+      },
+      beforeTestCase: (fn) => test.before.each(fn as Callback),
+      afterTestCase: (fn) => test.after.each(fn as Callback),
+    }
+
+    test.before(() => setTestRunnerIntegration(testRunnerCtx))
+
+    test.before.each((ctx) => {
+      testRunnerCtx.testInfo.suitName = ctx.__suite__ ? [ctx.__suite__] : []
+      testRunnerCtx.testInfo.testName = ctx.__test__
+    })
```

Comment:
Since ES modules can't really be monkey patched at the moment, I wonder if it would be helpful export this logic as its own function so ESM users can manually register their tests.

e.g.

```js
import { expect } from 'earljs'
import { registerUvuTest } from 'earljs/uvu'
import { test } from 'uvu'

test('some test', () => {
  expect(true).toEqual(true)
})

registerUvuTest(test, import.meta.url)
test.run()

```

and the monkeypatching functionality could go to `earljs/uvu/register`.